e2e-tests/cypress/support/page-objects/Discover.ts

```diff
@@ -83,6 +86,76 @@ export class Discover {
     }
     return openDataPageForKibanaBefore7_18_1();
   };
+
+  static navigateTo() {
+    cy.log('navigateTo');
+    KibanaNavigation.openKibanaNavigation();
+    cy.contains('Discover').click({force: true});
+    cy.wait(1000);
+  }
+
+  static selectDataView(dataViewName: string) {
+    cy.log('selectDataView');
+
+    // For Kibana 8.0.0 and above, we need to interact with the popover
+    if (semver.gte(getKibanaVersion(), '8.0.0')) {
+      cy.get('[data-test-subj="discover-dataView-switch-link"]').click(); // Open the popover
+
+      // Search for the data view within the popover
+      cy.get('.euiPopover__panel [role="combobox"] input') // Assuming this is the search input within the popover
+        .type(dataViewName);
+
+      // Wait for the options to filter and then select the desired data view
+      cy.wait(500); // Adjust if needed
+      cy.get(`.euiPopover__panel li:contains("${ dataViewName }")`)
+        .click({force: true});
+    } else {
+      // For older Kibana versions, use the previous logic
+      cy.get('[data-test-subj="indexPattern-switch-link"]').click();
+      cy.contains(dataViewName).click({force: true});
+    }
+  }
+
+  static generateCsvReport(timeout: number = 50000) {
+    cy.log('generateCsvReport');
+
+    // Click the share button to open the sharing options
+    cy.get('[data-test-subj=shareTopNavButton]').click();
+
+    // Click the "CSV Reports" option
+    cy.get('[data-test-subj=sharePanel-CSVReports]').click();
+
+    // Click the "Generate report" button
+    cy.get('[data-test-subj=generateReportButton]').click();
+
+    // Wait for the "Queued report for search" message to appear
+    cy.contains('Queued report for search', {timeout: timeout}).should('exist');
+
+    // Close the "Queued report for search" toast
+    cy.get('[data-test-subj="toastCloseButton"]').click();
+
+    // Wait for the "Queued report for search" message to disappear
+    cy.contains('Queued report for search', {timeout: timeout}).should('not.exist');
+    var triger: string;
+    if (semver.lte(getKibanaVersion(), '8.8.0')) {
+      triger = 'Created report for';
+    } else {
+      triger = 'CSV created for';
+    }
+    // Attempt to find the "Created report for" message within the timeout
```

Comment:
IMO it should work a little bit differently. In this test, we want to check if many reports running at the same time are going to work as expected. In the past, we have a problem with it. When we run reports one by one (waiting for one finish before starting another) everything was OK. But when we run many reports (and the reports generation was scheduled by Kibana to be executed by a different node than the node where the report was scheduled on), there was a problem. This is what we want to test in this task.

So, we want to run many reports one after another. Then we should go to the view where the reports are shown (management sth view) and wait there for the generation finish. When at least one fails, the test should fail.

Reply:
as we talked:
1. let's generate huge number of docs
2. let's cut off the services resources

Reply:
Fixed in `Self-generate invoices data, checking report generation process`